core/src/main/java/com/scalar/db/api/Put.java

```diff
@@ -779,41 +770,31 @@ public Optional<MutationCondition> getCondition() {
   }
 
   /**
-   * Returns whether implicit pre-read is enabled for this Put.
+   * Returns whether implicit pre-read is enabled for this Put. This is a utility method for
+   * Consensus Commit.
    *
    * @return whether implicit pre-read is enabled for this Put
+   * @deprecated As of release 3.15.0. Will be removed in release 5.0.0. Use {@link
+   *     ConsensusCommitOperationAttribute#isImplicitPreReadEnabled(Put)} instead
    */
+  @SuppressWarnings("InlineMeSuggester")
+  @Deprecated
   public boolean isImplicitPreReadEnabled() {
-    return implicitPreReadEnabled;
-  }
-
-  /**
-   * Sets whether implicit pre-read is enabled for this Put.
-   *
-   * @param implicitPreReadEnabled whether the implicit pre-read is enabled for this Put
-   */
-  Put setImplicitPreReadEnabled(boolean implicitPreReadEnabled) {
-    this.implicitPreReadEnabled = implicitPreReadEnabled;
-    return this;
+    return ConsensusCommitOperationAttribute.isImplicitPreReadEnabled(this);
```

Comment:
Just an idea: Moving these operations to `ConsensusCommitOperationAttribute`'s instance methods and replacing `ImmutableMap<String, String> attributes` with `ConsensusCommitOperationAttribute attributes` might be an option in the future.